sdk/client-core/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -283,4 +294,17 @@ public RequestOptions setResponseBodyMode(ResponseBodyMode responseBodyMode) {
 
         return this;
     }
+
+    /**
+     * Sets the {@link ClientLogger} used to log during the request and response.
+     *
+     * @param requestLogger The {@link ClientLogger} used to log during the request and response.
+     *
+     * @return The updated {@link RequestOptions} object.
+     */
+    public RequestOptions setRequestLogger(ClientLogger requestLogger) {
```

Comment:
Because this is in `RequestOptions`, we can probably just call the getter / setter `getLogger` / `setLogger`